src/Chapter3.hs

```diff
@@ -479,6 +520,61 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
+data Castle = Castle
+    {
+      castleName :: String
+    } deriving (Show)
+
+data ChLib =
+  Church |
+  Library
+
+data House =
+  One |
+  Two |
+  Three |
+  Four
+
+
+data City =
+          CityWithCastle
+          {
+            cityCastle :: Castle,
+            cityWall :: Bool,
+            cityChurchLib :: ChLib,
+            cityHouses :: [House]
+          } |
+          CityWithOutCastle
+          {
+            cityChurchLib :: ChLib,
+            cityHouses :: [House]
+          }
+
+howManyInHouse :: House -> Int
+howManyInHouse house = case house of One -> 1
+                                     Two -> 2
+                                     Three -> 3
+                                     Four -> 4
+
+howManyInHouses :: [House] -> Int
+howManyInHouses houses = foldr (+) 0 (map (\x-> (howManyInHouse x)) houses)
```

Comment:
`foldr (+) 0` is the standard function called `sum` and you can eta-reduce the lambda in `map` :recycle: 
```suggestion
howManyInHouses houses = sum (map howManyInHouse houses)
```